File: commonlib/src/main/java/com/sixiangtianxia/commonlib/utils/MyUtils.java
package com.sixiangtianxia.commonlib.utils;

import android.annotation.SuppressLint;
import android.app.Application;
import android.support.annotation.NonNull;


/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 16/12/08
 *     desc  : Utils初始化相关
 * </pre>
 */
public final class MyUtils {

    @SuppressLint("StaticFieldLeak")
    private static Application sApplication;



    private MyUtils() {
        throw new UnsupportedOperationException("u can't instantiate me...");
    }

    /**
     * 初始化工具类
     *
     * @param app 应用
     */
    public static void init(@NonNull final Application app) {
        MyUtils.sApplication = app;
        new RudenessScreenHelper(app,1080).activate();

    }

    /**
     * 获取Application
     *
     * @return Application
     */
    public static Application getApp() {
        if (sApplication != null) return sApplication;
        throw new NullPointerException("u should init first");
    }

}

File: commonlib/src/main/java/com/sixiangtianxia/commonlib/utils/MyUtils.kt
package com.sixiangtianxia.commonlib.utils

import android.annotation.SuppressLint
import android.app.Application


/**
 * <pre>
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 16/12/08
 * desc  : Utils初始化相关
</pre> *
 */
class MyUtils private constructor() {



    init {
        throw UnsupportedOperationException("u can't instantiate me...")
    }

    companion object {

        @SuppressLint("StaticFieldLeak")
        private var sApplication: Application? = null

        /**
         * 初始化工具类
         *
         * @param app 应用
         */
        fun init(app: Application) {
            MyUtils.sApplication = app
//            RudenessScreenHelper(app, 1080f).activate()

        }

        /**
         * 获取Application
         *
         * @return Application
         */
        val app: Application
            get() {
                if (sApplication != null) return sApplication!!
                throw NullPointerException("u should init first")
            }
    }

}
